### PR TITLE
Upgrade prometheus-operator to 0.31.0 and improve requests/limits for…

### PR DESCRIPTION
**Pull request by Weil0ng**
… prometheus

**What type of PR is this?**
backport

**What this PR does / why we need it**:
Need `podMonitor` support with release-1.18

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by Weil0ng** [comment]
@jprzychodzen suggested we backport this to `release-1.18` so we can have `podMonitor` support required by our tests

File: clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml [hunk 1]
```diff
@@ -13,18 +13,26 @@ spec:
     - name: alertmanager-main
       namespace: monitoring
       port: web
+  logLevel: debug
```
Comment (jprzychodzen) [inline clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml:16]: It seems that there is an idea to revert this logLevel from original PR - https://github.com/kubernetes/perf-tests/commit/c4c88273493d802a9ec22cdbde0efe8f4308b531#commitcomment-46472916

Could you remove this logLevel line?

Comment (Weil0ng) [reply clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml]: Done.

**Pull request merged**